src/main/java/uk/gov/hmcts/reform/sendletter/services/encryption/PgpEncryptionUtil.java

```diff
@@ -44,43 +48,49 @@ private PgpEncryptionUtil() {
      * @param withIntegrityCheck Sets whether or not the resulting encrypted data will be protected
      *                           using an integrity packet.
      * @return PGP encrypted byte array
-     * @throws IOException  if an error occurs writing stream header information to the provider
-     *                      output stream.
-     * @throws PGPException if an error occurs initialising PGP encryption for the configured
-     *                      encryption method.
      */
     public static byte[] encryptFile(
         byte[] inputFile,
         String inputFileName,
         PGPPublicKey pgpPublicKey,
         boolean withIntegrityCheck
-    ) throws IOException, PGPException {
-        Security.addProvider(new BouncyCastleProvider());
-
-        ByteArrayOutputStream byteArrayOutputStream =
-            compressAndWriteFileToLiteralData(
-                inputFile,
-                inputFileName
-            );
-
-        PGPEncryptedDataGenerator encryptedDataGenerator =
-            prepareDataEncryptor(
-                pgpPublicKey,
-                withIntegrityCheck
-            );
-
-        return writeEncryptedDataToOutputStream(byteArrayOutputStream, encryptedDataGenerator);
+    ) {
+        try {
+            Security.addProvider(new BouncyCastleProvider());
+
+            ByteArrayOutputStream byteArrayOutputStream =
+                compressAndWriteFileToLiteralData(
+                    inputFile,
+                    inputFileName
+                );
+
+            PGPEncryptedDataGenerator encryptedDataGenerator =
+                prepareDataEncryptor(
+                    pgpPublicKey,
+                    withIntegrityCheck
+                );
+
+            return writeEncryptedDataToOutputStream(byteArrayOutputStream, encryptedDataGenerator);
+        } catch (IOException | PGPException exc) {
+            log.error("Error encrypting file", exc);
+            throw new UnableToPgpEncryptZipFileException(exc);
+        }
     }
 
     /**
      * Returns raw key bytes as a Bouncy Castle PGP public key.
      */
-    public static PGPPublicKey loadPublicKey(byte[] data) throws IOException {
-        return lookupPublicSubkey(
-            new BcPGPPublicKeyRing(
-                getDecoderStream(new ByteArrayInputStream(data))
-            )
-        ).orElseThrow(() -> new UnableToLoadPgpPublicKeyException("PGP Public key object could not be constructed"));
+    public static PGPPublicKey loadPublicKey(byte[] data) {
+        try {
+            return lookupPublicSubkey(
+                new BcPGPPublicKeyRing(
+                    getDecoderStream(new ByteArrayInputStream(data))
+                )
+            ).orElseThrow(() -> new UnableToLoadPgpPublicKeyException(null));
```

Comment:
Unrelated to this change, but should UnableToLoadPgpPublicKeyException exist as an Exception? It seems more like an assertion failure that should immediately crash the app, it isn't an exceptional condition that can be handled.

Reply:
Same here, could do with just `UnableToLoadPublicKeyException`. Not even pgp related then. specifically for any attempt to load public key